### PR TITLE
[OpaquePtr] Emit target extension types for sampled images for SPIR-V.

### PR DESCRIPTION
**Pull request by jcranmer-intel**

### CODE REVIEW

**Comment by jcranmer-intel** [comment]
This adds the changes in https://github.com/intel/llvm/blob/d5a273111ae816a5471cdc1fd0c64e956fe02b50/clang/lib/CodeGen/CGOpenCLRuntime.cpp#L73-L81 to the SPIR-V target extension type generation.

**Comment by smanna12** [comment]
@jcranmer-intel Do you have any test for this change?

**Comment by jcranmer-intel** [comment]
> @jcranmer-intel Do you have any test for this change?

No, I haven't prepared any specific tests for this change yet.

File: clang/lib/CodeGen/TargetInfo.cpp [hunk 1]
```diff
@@ -11252,6 +11252,12 @@ llvm::Type *CommonSPIRTargetCodeGenInfo::getOpenCLType(CodeGenModule &CGM,
 #define IMAGE_TYPE(ImgType, Id, SingletonId, Access, Suffix)                   \
     case BuiltinType::Id:                                                      \
       return getSPIRVImageType(Ctx, "spirv.Image", #ImgType, AQ_##Suffix);
+#include "clang/Basic/OpenCLImageTypes.def"
+#define IMAGE_TYPE(ImgType, Id, SingletonId, Access, Suffix)                   \
```
Comment (Fznamznon) [inline clang/lib/CodeGen/TargetInfo.cpp:11256]: Shouldn't we do that in the community?

Comment (jcranmer-intel) [reply clang/lib/CodeGen/TargetInfo.cpp:11256]: The `Sampled*` types haven't been upstreamed yet, which means this change itself can't be upstreamed.

**Review by Fznamznon — APPROVED** [review]

**Comment by dm-vodopyanov** [comment]
> Hi, what's blocking merging this PR at this point?

@jcranmer-intel please fix clang-format issue, `SYCL Pre Commit / lint (pull_request_target)` is failing.

**Review by smanna12 — APPROVED** [review]

**Comment by jcranmer-intel** [comment]
> > Hi, what's blocking merging this PR at this point?
> 
> @jcranmer-intel please fix clang-format issue, `SYCL Pre Commit / lint (pull_request_target)` is failing.

In this case, I believe the existing change is correct formatting, and the clang-format suggested fix would cause the code to become incorrectly formatted--it moves the case statements left by one indentation level, before the switch statements.

**Comment by dm-vodopyanov** [comment]
> > > Hi, what's blocking merging this PR at this point?
> > 
> > 
> > @jcranmer-intel please fix clang-format issue, `SYCL Pre Commit / lint (pull_request_target)` is failing.
> 
> In this case, I believe the existing change is correct formatting, and the clang-format suggested fix would cause the code to become incorrectly formatted--it moves the case statements left by one indentation level, before the switch statements.

If clang-format suggests wrong formatting for some block of code, please wrap this block with
```
// clang-format off
...
// clang-format on
```

**Pull request merged**